distribution/lib/Standard/Base/0.0.0-dev/src/Network/URI.enso

```diff
@@ -128,7 +128,26 @@ type URI
 
              example_path = Examples.uri.path
     path : Text | Nothing
-    path self = self.internal_uri.getPath
+    path self = handle_resolve_errors <|
+        self.to_java_representation.getPath
+
+    ## GROUP Calculations
+       Extends the path of this URI with an additional segment.
+
+       Arguments:
+       - segment: The segment to add to the path.
+         The `segment` may contain multiple segments separated by `/` characters.
+         Any other characters that are not normally permitted in a URI will be
+         percent-encoded.
+    / : Text -> URI
+    / self segment:Text =
```

Comment:
not saying it should, but we could add pre the query string part

Reply:
I don't think I understand what you mean, could you please clarify?

Reply:
Isn't what you mean the thing that is checked in this test?
```
        Test.specify "should keep existing query arguments intact when extending the path" <|
            uri0 = URI.parse "https://example.com/path?a=b&c=d"
            uri1 = uri0 / "x" / "y"
            uri2 = uri1 . add_query_argument "e" "f"
            uri3 = uri2 / "z"

            uri3.path . should_equal "/path/x/y/z"
            uri3.query . should_equal "a=b&c=d&e=f"
            uri3.to_text . should_equal "https://example.com/path/x/y/z?a=b&c=d&e=f"
```

or is that sth else?